Add FastAPI and SQLModel to third-party tests (#11044)
pydantic/pydantic: docs/contributing.md

<a href="https://pydantic.dev"><img src="https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/pydantic/pydantic/main/docs/badge/v2.json" alt="Pydantic Version 2" style="max-width:100%;"></a>
```

> ## Adding your library as part of Pydantic tests

Author: ```suggestion
## Adding your library as part of Pydantic's third party test suite
```

To be able to identify regressions early during development, Pydantic runs tests on various third-party projects

> using Pydantic. Are you maintaining an open source project and are you heavily relying on Pydantic? If you match
> some of the following criteria:

Author: ```suggestion
using Pydantic. We consider adding support for testing new open source projects (that rely heavily on Pydantic) if your said project matches some of the following criteria:
```

- The project is actively maintained.

> - Making use of Pydantic internals (e.g. relying on the [`BaseModel`][pydantic.BaseModel] metaclass, typing utilities).

Author: ```suggestion
- The project makes use of Pydantic internals (e.g. relying on the [`BaseModel`][pydantic.BaseModel] metaclass, typing utilities).
```

> - The project is popular enough (although small projects can still be included depending on how Pydantic is being used).

Author: I feel like we should quantify this (some # of stars, downloads)

Reviewer: The thing is it really depends on how Pydantic is used. We can have project with 100k stars but if it is only defining simple models (e.g. models for an API client, and types are well contained) then there's no interest in testing it. On the other hand, a project with 200 stars defining complex custom types is more valuable. So it is hard to quantify :/

> - The project CI is simple enough to be ported into Pydantic's workflow.

Author: ```suggestion
- The project CI is simple enough to be ported into Pydantic's testing workflow.
```

> The project can be tested as part of Pydantic's development process. You can [open feature request][open feature request]
> to discuss the inclusion of your project.

Author: ```suggestion
If your project meets some of these criteria, you can [open feature request][open feature request]
to discuss the inclusion of your project.
```

Or something like that, feel free to workshop, seems a little redundant right now with the above leading sentence
